python/src/foxglove_websocket/server/__init__.py

```diff
@@ -163,6 +163,7 @@ def __init__(
         metadata: Optional[Mapping[str, str]] = None,
         session_id: Optional[str] = None,
         logger: logging.Logger = _get_default_logger(),
+        server_kwargs: Dict[str, Any] = {"compression": None},
```

Comment:
Is there something we can do to pass through the kwarg types from the serve function so autocomplete and type checking work properly?

Reply:
Apparently you can inspect the signature of a function/object (e.g. serve) such as
```
>>> from websockets.server import serve
>>> import inspect
>>> inspect.signature(server)
<Signature (ws_handler: 'Union[Callable[[WebSocketServerProtocol], Awaitable[Any]], Callable[[WebSocketServerProtocol, str], Awaitable[Any]]]', host: 'Optional[Union[str, Sequence[str]]]' = None, port: 'Optional[int]' = None, *, create_protocol: 'Optional[Callable[[Any], WebSocketServerProtocol]]' = None, logger: 'Optional[LoggerLike]' = None, compression: 'Optional[str]' = 'deflate', origins: 'Optional[Sequence[Optional[Origin]]]' = None, extensions: 'Optional[Sequence[ServerExtensionFactory]]' = None, subprotocols: 'Optional[Sequence[Subprotocol]]' = None, extra_headers: 'Optional[HeadersLikeOrCallable]' = None, process_request: 'Optional[Callable[[str, Headers], Awaitable[Optional[HTTPResponse]]]]' = None, select_subprotocol: 'Optional[Callable[[Sequence[Subprotocol], Sequence[Subprotocol]], Subprotocol]]' = None, ping_interval: 'Optional[float]' = 20, ping_timeout: 'Optional[float]' = 20, close_timeout: 'Optional[float]' = None, max_size: 'Optional[int]' = 1048576, max_queue: 'Optional[int]' = 32, read_limit: 'int' = 65536, write_limit: 'int' = 65536, **kwargs: 'Any') -> 'None'>
```
which you can very problably use somethow as a type for `server_kwarg`.

However, some of the args are set manually (`self.host`, `self.port`, ...) and not part of these `server_kwargs`.